pkg/reconciler/instances/istio/actions/performer.go

```diff
@@ -75,52 +77,66 @@ type DataPlaneVersion struct {
 	IstioVersion string `json:"IstioVersion,omitempty"`
 }
 
+type CommanderResolver interface {
+	GetCommander(version istioctl.Version) (istioctl.Commander, error)
+}
+
 //go:generate mockery --name=IstioPerformer --outpkg=mock --case=underscore
 // IstioPerformer performs actions on Istio component on the cluster.
 type IstioPerformer interface {
 
-	// Install Istio on the cluster.
-	Install(kubeConfig, manifest string, logger *zap.SugaredLogger) error
+	// Install Istio on the cluster. Tries to find best matching istioctl binary for the 'requestedVersion'.
+	Install(kubeConfig, istioChart, requestedVersion string, logger *zap.SugaredLogger) error
 
 	// PatchMutatingWebhook configuration.
 	PatchMutatingWebhook(kubeClient kubernetes.Client, logger *zap.SugaredLogger) error
 
-	// Update Istio on the cluster.
-	Update(kubeConfig, manifest string, logger *zap.SugaredLogger) error
+	// Update Istio on the cluster. Tries to find best matching istioctl binary for the 'requestedVersion'.
+
+	Update(kubeConfig, istioChart, requestedVersion string, logger *zap.SugaredLogger) error
 
-	// ResetProxy of all Istio sidecars on the cluster.
-	ResetProxy(kubeConfig string, version IstioVersion, logger *zap.SugaredLogger) error
+	// ResetProxy of all Istio sidecars on the cluster. The proxyImageVersion parameter controls the Istio proxy version, it always adds "-distroless" suffix to the provided value.
+	ResetProxy(kubeConfig string, proxyImageVersion string, logger *zap.SugaredLogger) error
 
-	// Version of Istio on the cluster.
-	Version(workspace chart.Factory, branchVersion string, istioChart string, kubeConfig string, logger *zap.SugaredLogger) (IstioVersion, error)
+	// Reports status of Istio versions on the cluster.
```

Comment:
Docs should start with the struct / interface / function etc. name, in this case with `Version`.